lib/jekyll/readers/theme_data_reader.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Jekyll
+  class ThemeDataReader < Jekyll::DataReader
+    attr_reader :site, :content
+
+    def read(dir)
+      return unless site.theme&.data_path
+
+      base = site.in_theme_dir(dir)
```

Comment:
I believe theme directories are standardized and not configurable. I would feel more comfortable releasing this with only one name for the theme data dir. Our intention was always to make it so a theme could be a Jekyll site, so I'd recommend the default `_data`. So `site.in_theme_dir(site.theme.data_path)`, I think?